### PR TITLE
Adjust for new hse-java package name

### PR DESCRIPTION
**Pull request by tristan957**
Signed-off-by: Tristan Partin <tpartin@micron.com>

### CODE REVIEW

**Comment by tristan957** [comment]
@bhaveshvasandani next time we bump the version on hse-java, we need to remember to bump the required version for ycsb.

**Comment by tristan957** [comment]
Note to self, merge https://github.com/hse-project/hse-ycsb/pull/18 first.

File: hse/pom.xml [hunk 1]
```diff
@@ -38,9 +38,9 @@ This code is derived from the ycsb project.
       <version>0.10.3</version>
     </dependency>
     <dependency>
-      <groupId>com.micron.hse_project</groupId>
+      <groupId>io.github.hse_project</groupId>
```
Comment (alexttx) [inline hse/pom.xml:41]: Educate me.  Why are we making this change?

Comment (tristan957) [reply hse/pom.xml:41]: Essentially the issue is that OSSRH, the repo we will publish to, requires you to verify domain ownership. One way to do this is through DNS records.

The chances we get a top level DNS record for micron.com? Probably less than 0.

Luckily OSSRH allows us to use our GitHub website for the package name.

This change happened in coordination with Steve and Bhavesh which is why you are just now finding out about it.

**Review by alexttx — APPROVED** [review]

**Pull request merged**